DRILL-4314: Unit Test Framework can support schema check

hsuanyi opened the pull request.

jacques-n commented:
@jaltekruse, can you review?

jaltekruse commented:
Can do

jaltekruse commented on exec/java-exec/src/test/java/org/apache/drill/TestBuilder.java:
```diff
@@ -235,6 +240,20 @@ public CSVTestBuilder csvBaselineFile(String filePath) {
 
+  public SchemaTestBuilder schemaBaseLine(List<Pair<SchemaPath, TypeProtos.MajorType>> expectedSchema) {
+    assert expectedSchema != null : "The expected schema can be provided once";
+    assert baselineColumns == null : "The column information should be captured in expected schem, not baselineColumns";
```
Could you add a few tests to the TestFrameworkTest that hit these conditions? It would be good if future changes to the test builder that may cause some currently-valid test configurations to fail (or currently invalid configurations to be allowed) would be detected by the tests for the framework itself.

jaltekruse commented:
@hsuanyi Other than the small comment this looks good to me +1

hsuanyi commented:
@jaltekruse, thanks for the review. I addressed the comments.
Besides, I also did a rebasing and clean run.  You can commit it in at your convenience. Thanks!
